### PR TITLE
TransportParams support in ClientOptions

### PR DESCRIPTION
**Pull request by ikurek**
This should close https://github.com/ably/ably-flutter/issues/108. I've found out that this feature was implemented on Dart side, but it wasn't implemented in iOS and Android bridges, probably because of differences in implementation of these methods for each platform (iOS requires `NSDictionary` and Android requires `Param[]` array).  I've implemented bridges on each platform and now it seems to work fine 🙂

### CODE REVIEW

**Review by KacperKluka — CHANGES_REQUESTED** [review]

File: android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java [hunk 1]
```diff
@@ -371,6 +371,18 @@ private TokenDetails decodeTokenDetails(Map<String, Object> jsonMap) {
     return o;
   }
 
+  private Param[] decodeTransportParams(Map<String, String> jsonMap) {
+    if (jsonMap == null) return null;
+    // It's not possible to initialize the array here, because that way,
+    // Params will have a (null, null) entry, so we need to initialize it later with `Params.set`
+    Param[] o = null;
```
[the remark this answers is not in the document]
Comment (ikurek) [reply android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java:378]: We definitely should, but if you look at the file, these variables are called `o` in every codec method 😞  I've decided to stick to the convention used previously, should I change it in that case? 🤔

[the remark this answers is not in the document]
Comment (KacperKluka) [reply android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java:378]: I'd say let's make it the first one which has some meaningful names :grin: There's no need to align with a bad practice IMO :wink:

[the remark this answers is not in the document]
Comment (ikurek) [reply android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java]: Fixed in https://github.com/ably/ably-flutter/pull/364/commits/2e40e666f277a9dc24bc51ef0cf24a72cb3fa851

**Review by KacperKluka — APPROVED** [review]
LGTM :+1:

**Review by lukasz-szyszkowski — APPROVED** [review]

**Pull request merged**